### ISSUE TITLE
Add test coverage for StandardOleMarshalObject

### ISSUE THREAD

**Issue by ericstj** [dotnet/corefx#32494]
See https://github.com/dotnet/corefx/pull/32491

We should add test coverage for this newly ported type.

@luqunl do you know of a trick to route a call through COM using purely managed code (eg: an RCW(CCW(managedobject)))

If we can do that, then I can imagine we could just call a COM-exposed method on the StandardOleMarshalObject derived type from a different thread and verify that it runs on the main thread.

/cc @Anipik

**Comment by luqunl** [comment]
Yes, You can use ICustomQueryInterface to trick interop --- the StandardOleMarshalObject derived type should implements ICustomQueryInterface to tell interop this is a native object.

``
    class MyStandardOleMarshalObject : StandardOleMarshalObject, ICustomQueryInterface
    {

        static readonly Guid IID_IMarshal = new Guid("00000003-0000-0000-C000-000000000046");
        static readonly Guid IID_IManagedObject = new Guid("C3FCC19E-A970-11d2-8B5A-00A0C9B7C9C4");

        CustomQueryInterfaceResult ICustomQueryInterface.GetInterface(ref Guid iid, out IntPtr ppv)
        {
            if (iid == IID_IMarshal ||
                iid == IID_IManagedObject
                )
            {
                ppv = IntPtr.Zero;
                return CustomQueryInterfaceResult.Failed;
            }

            ppv = IntPtr.Zero;
            return CustomQueryInterfaceResult.NotHandled;
        }

``
